test/markup/markdown/bold_italics.txt

```diff
@@ -6,13 +6,19 @@ __Bold__
 __Bold *then italic*__
 **Bold _then italic_**
 **Bold and <br/>**
-**_[this](https://google.com)_**
+__*[this](https://google.com)*__
```

Comment:
The rule to match underscores checks if there is a space or newline before it. This is how it prevents matching underscores inside of words. Since the containing bold has already matched the asterisks. The underscore rule can't match the space/newline. A workarounds might be to add a rule specifically for `**_`, `___` and `*__` but its a tradeoff in complexity. Can add rules for that if you'd like.

Reply:
I'm not sure I follow. Are you saying you removed these tests because they now fail?  That's not what we need to be doing?

Reply:
If I make changes so that the parser is more correct. It's too complex and you'd rather have something that is simpler and "good enough". If I go the simpler route, certain edge cases are going to regress. But other, more common use cases will now be improved. Its a tradeoff.

Reply:
Ok, it's certainly _possible_ we may need tradeoffs.   Yet please don't hide these decisions by silently removing tests...   If a change is deciding to make a tradeoff then it should be mentioned explicitly in the PR notes (with examples) what that comprise is, what the thinking is behind the change, etc... ie if you're breaking an edge case and improving a common case that thinking should be outlined in the PR notes... so that it can be easily reviewed to decide if the tradeoff is an acceptable one or not.